### PR TITLE
[QC-357] Move Data Sampling from Framework to Utilities

### PR DESCRIPTION
**Pull request by knopers8**
Needs https://github.com/AliceO2Group/QualityControl/pull/457

### CODE REVIEW

File: Utilities/DataSampling/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                         src/DataSamplingConditionCustom.cxx
+                         src/DataSamplingConditionNConsecutive.cxx
+                         src/DataSamplingConditionPayloadSize.cxx
+                         src/DataSamplingConditionRandom.cxx
+                         src/DataSamplingHeader.cxx
+                         src/DataSamplingPolicy.cxx
+                         src/DataSamplingReadoutAdapter.cxx
+                         src/Dispatcher.cxx
+
+  PUBLIC_LINK_LIBRARIES O2::Framework O2::DataSampling)
```
Comment (Barthelemy) [inline Utilities/DataSampling/CMakeLists.txt:26]: I am not sure about declaring DataSampling as dependency of DataSampling.

**Review by Barthelemy — COMMENTED** [review]
I did not spot mistakes I am just wondering about 1 piece in the CMakeLists.txt

File: Utilities/DataSampling/CMakeLists.txt (hunk 1 above)
Comment (knopers8) [reply Utilities/DataSampling/CMakeLists.txt:26]: indeed, this is wrong

**Review by Barthelemy — DISMISSED** [review]

**Review by ktf — DISMISSED** [review]

**Comment by knopers8** [comment]
Great, we only need a QC release and green checks.

**Comment by ktf** [comment]
tests were not running because of the [WIP].

**Comment by Barthelemy** [comment]
ok,so we need QC v1.1 in alidist for this to work

**Comment by knopers8** [comment]
@Barthelemy @ktf The checks are mostly green aside from the gpu check, where the error is common in all PRs. This PR does not touch GPU code, so it shouldn't break anything in this build. Can we proceed with merging this?

**Comment by knopers8** [comment]
thanks

**Pull request merged**